test/common/network/cidr_range_test.cc

```diff
@@ -0,0 +1,240 @@
+#include "envoy/common/exception.h"
+
+#include "common/network/cidr_range.h"
+#include "common/network/address_impl.h"
+
+#include <sys/un.h>
+
+// We are adding things into the std namespace.
+// Note that this is technically undefined behavior!
+namespace std {
+
+// Pair
+template <typename First, typename Second>
+std::ostream& operator<<(std::ostream& out, const std::pair<First, Second>& p) {
```

Comment:
Hmm. Well, most of those appear to be unused, only the two Buffer variants appear to have callers.
The data type is a pair<pair, pair>, so the template gets expanded 3 times (really two, the two inner types are the same), which is awfully convenient. Also, PrintTo can't be used after EXPECT*, so it doesn't seem a great solution here.